endpoint.go

```diff
@@ -637,6 +639,14 @@ func EndpointOptionGeneric(generic map[string]interface{}) EndpointOption {
 	}
 }
 
+// CreateOptionIpam function returns an option setter for the ipam configuration for this endpoint
+func CreateOptionIpam(prefAddress net.IP, ipamOptions map[string]string) EndpointOption {
```

Comment:
@jc-m Nothing against the multiple addresses. Just wanted to go one step at the time. As we know each new functionality comes with its own unforeseen bugs.

My thinking is to get the preferred ip functionality in ASAP so that we have enough time to discover any corner case. If things show to be smooth, extending to multiple addresses will be a safe, quick and easy change.